model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/realm/HotRodRealmEntity.java

```diff
@@ -271,6 +271,9 @@ public interface HotRodRealmEntitySchema extends GeneratedSchema {
     @ProtoField(number = 77)
     public Set<String> supportedLocales;
 
+    @ProtoField(number = 78)
+    public Boolean emailFormatAsUsernameAllowed;
```

Comment:
I'm not sure what's the best approach for creating realm flags here. For the JPA, we use inner attributes due to avoid schema updates and following migrations. 

@martin-kanis Maybe even for ProtoBufs, it'd be good to use some inner attributes, right? As it'd be quite tricky in the future to remove some fields from here, right?

Reply:
@mabartos Good question! I discussed it briefly with @mhajas and we incline towards adding new fields to attributes whenever possible. Also for `map-jpa`, it maybe belongs to attributes but I would let @vramik to verify it.
This approach however creates inconsistency mainly in `HotRodRealmEntity` where legacy fields are directly defined in entity while new fields will be added to attributes. @hmlnarik Is this OK or do we want to have a wider discussion about this in the storage team?

Reply:
For `map-jpa` it could remain as it is implemented in this PR at the moment. The field ends up as a part of the json column.

Reply:
The right approach in cases like this is to leverage realm attributes instead and don't modify the entity structure.

Reply:
Thanks! That's very valuable information.